test/distributed/test_functional_api.py

```diff
@@ -234,6 +235,26 @@ def test_all_reduce_coalesced_eager(self):
         self.assertEqual(res[0], t0 * 4)
         self.assertEqual(res[1], t1 * 4)
 
+    def test_all_gather_into_tensor_coalesced(self):
+        tensors = [torch.ones([4], device="cpu"), torch.ones([4], device="cpu") + 1]
+        mesh = dt.DeviceMesh("cpu", torch.arange(4))
+
+        res = ft_c.all_gather_into_tensor_coalesced(tensors, mesh)
+        self.assertEqual(2, len(res))
+        self.assertEqual(torch.ones([4 * dist.get_world_size()]), res[0])
+        self.assertEqual(torch.ones([4 * dist.get_world_size()]) + 1, res[1])
+
+
+    @skip_if_lt_x_gpu(1)
+    def test_all_gather_into_tensor_coalesced_cuda(self):
```

Comment:
looks like the test cases are identical to the cpu case, can we some how merge them into 1? i.e. device_type = "cuda" if torch.cuda.is_available() and torch.cuda.device_count() < world_size else "cpu"

Reply:
I'm not a big fan of this kind of design that depends on the environment because it's not easily discoverable.
I'm going to unify it with parametrization though.